src/test/java/de/zalando/aruha/nakadi/validation/EventBodyMustRespectSchema.java

```diff
@@ -0,0 +1,233 @@
+
+    public static final String NAME = "event-body-must-respect-schema";
+
+    private static final Function<OverrideDefinition, QualifiedJSONSchemaValidator> toQualifiedJSONSchemaValidator =
+        t -> {
+        final JSONSchemaValidator jsv = new JSONSchemaValidator(t.getEffectiveSchema());
+        return new QualifiedJSONSchemaValidator(t.getQualifier(), jsv);
+    };
+
+    @Override
+    public EventValidator materialize(final EventType eventType, final ValidationStrategyConfiguration vsc) {
+
+        final JSONSchemaValidator defaultSchemaValidator = new JSONSchemaValidator(eventType.getEventTypeSchema()
+                    .getSchema());
+        if (vsc.getAdditionalConfiguration() == null) {
+            return defaultSchemaValidator;
+        }
+
+        final ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JsonOrgModule());
+
+        final Function<OverrideDefinition, OverrideDefinition> enhanceWithQualifiedSchema = definition -> {
+            final JSONObject schema = eventType.getEventTypeSchema().getSchema();
+            final JSONObject copy = new JSONObject(schema, JSONObject.getNames(schema));
+
+            definition.getIgnoredProperties().forEach(new Consumer<String>() {
+                    @Override
+                    public void accept(final String name) {
+                        copy.getJSONObject("properties").remove(name);
+
+                        final JSONArray array = copy.getJSONArray("required");
+                        final int idx = findElement(array, name);
+                        if (idx > -1) {
+                            array.remove(idx);
+                        }
+                    }
+                });
+            definition.setEffectiveSchema(copy);
+            return definition;
+        };
+
+        final List<QualifiedJSONSchemaValidator> qualifiedValidators = mapper.convertValue(
+                                                                                 vsc.getAdditionalConfiguration(),
+                                                                                 SchemaValidationOverrides.class)
+                                                                             .getOverrides().stream()
+                                                                             .map(enhanceWithQualifiedSchema)
+                                                                             .map(toQualifiedJSONSchemaValidator)
+                                                                             .collect(Collectors.toList());
+
+        return new QualifiedJSONSchemaValidationChain(qualifiedValidators, defaultSchemaValidator);
+    }
+
+    public static int findElement(final JSONArray array, final String element) {
+
+        for (int i = 0; i < array.length(); i++) {
+            if (element.equals(array.getString(i))) {
+                return i;
+
+            }
+        }
+
+        return -1;
```

Comment:
I think it's actually a perfect place to use Optional<Integer> as a return type to avoid returning -1